ipaserver/install/server/replicainstall.py

```diff
@@ -918,47 +918,55 @@ def install(installer):
 
 
 def ensure_enrolled(installer):
-    config = installer._config
+    # Prepare options for the installer script
+    args = [paths.IPA_CLIENT_INSTALL, "--no-ntp"]
+    stdin = None
+
+    if installer.unattended:
+        args.append("--unattended")
+    if installer.domain_name:
+        args.extend(["--domain", installer.domain_name])
+    if installer.server:
+        args.extend(["--server", installer.server])
+    if installer.realm_name:
+        args.extend(["--realm", installer.realm_name])
+    if installer.host_name:
+        args.extend(["--hostname", installer.host_name])
+    if installer.password:
+        args.extend(["--password", installer.password])
+    else:
+        if installer.principal:
+            args.extend(["--principal", installer.principal])
+        if installer.admin_password:
+            if installer.principal is None:
+                raise ScriptError("The --admin-password option must be used "
+                                  "with the --principal option.")
+            if installer.unattended:
+                # Don't add the password to the options in unattended mode
+                # ==> it would also appear in the client install logs
+                stdin = installer.admin_password
+            else:
+                args.extend(["--password", installer.admin_password])
```

Comment:
This way a password is logged in replica install log, which is undesirable. There might be some logging option to clear the password.